spec/classes/plugin_migration_spec.rb

```diff
@@ -0,0 +1,36 @@
+require 'spec_helper'
+
+migration_config = <<-EOL
+PULP2_MONGODB = {
+    "name": "pulp_database",
+    "seeds": "localhost:27017",
+    "ssl": "False"
+}
+EOL
+
+describe 'pulpcore::plugin::migration' do
+  on_supported_os.each do |os, os_facts|
+    context "on #{os}" do
+      let(:facts) { os_facts }
+
+      it { is_expected.to compile.with_all_deps }
+      it { is_expected.to contain_pulpcore__plugin('migration') }
+
+      context 'with pulpcore' do
+        let(:pre_condition) { 'include pulpcore' }
+
+        it do
+          is_expected.to compile.with_all_deps
+          is_expected.to contain_pulpcore__plugin('migration')
+            .that_subscribes_to('Class[Pulpcore::Install]')
+          is_expected.to contain_pulpcore__plugin('migration')
```

Comment:
This shouldn't be needed. You can chain them

Reply:
@ekohl I noticed that some were not getting executed when I did this. However removing the chaining did not resolve it. Is something excluding redundant tests?

Reply:
@ekohl after trying a few different approaches, I believe that the chaining works however the full output is suppressed. Therefore I have reverted to the chained approach.